src/mlpack/methods/decision_tree/decision_tree.hpp

```diff
@@ -42,6 +43,7 @@ template<typename FitnessFunction = GiniGain,
          template<typename> class NumericSplitType = BestBinaryNumericSplit,
          template<typename> class CategoricalSplitType = AllCategoricalSplit,
          typename DimensionSelectionType = AllDimensionSelect,
+         typename MDLPenaltyType = NoPenalty<FitnessFunction>,
```

Comment:
NoPenality is not declared before it is used, which will cause an error. So, declare the class before its use.
And as you are adding new template parameter, be sure to change all the occurrences of decisionTree instances in mlpack, for example ID3decisionStump which present at the end of decision_tree.hpp file. I recommend you to shift the MDLPenaltyType to the end of template parameters i.e. after NoRecursion to avoid changing other instances. You can ask for advice in mlpack community channel, whether to keep the MDLPenaltyType at the last second or last parameter.